proc/threads.c

```diff
@@ -1462,14 +1468,8 @@ int threads_sigpost(process_t *process, thread_t *thread, int sig)
 	}
 	hal_spinlockSet(&threads_common.spinlock, &sc);
 
-	if (thread != NULL && hal_cpuPushSignal(thread->kstack + thread->kstacksz, thread->process->sighandler, sig) != EOK) {
+	if (thread != NULL) {
 		thread->sigpend |= sigbit;
-
-		if (sig == signal_segv || sig == signal_illegal) {
-			/* If they can't handle those right away, kill */
-			kill = 1;
-			_proc_threadExit(thread);
-		}
```

Comment:
Hmm, we need to handle insta-kill procedure somewhere. Did you checked that PF is able to kill guilty process (it had a problems on the current code too, because we determined that the process is able to handle the signal when in fact it was not)